lib/matplotlib/axes/_base.py

```diff
@@ -3910,10 +3910,32 @@ def format_ydata(self, y):
 
     def format_coord(self, x, y):
         """Return a format string formatting the *x*, *y* coordinates."""
+        twins = self._twinned_axes.get_siblings(self)
+        if len(twins) == 1:
+            return "x={} y={}".format(
+                "???" if x is None else self.format_xdata(x),
+                "???" if y is None else self.format_ydata(y))
+        x_twins = []
+        y_twins = []
+        # Retrieve twins in the order of self.figure.axes to sort tied zorders (which is
+        # the common case) by the order in which they are added to the figure.
+        for ax in self.figure.axes:
+            if ax in twins:
+                if ax is self or ax._sharex is self or self._sharex is ax:
+                    x_twins.append(ax)
+                if ax is self or ax._sharey is self or self._sharey is ax:
+                    y_twins.append(ax)
+        get_zorder = attrgetter("zorder")
+        x_twins.sort(key=get_zorder)
+        y_twins.sort(key=get_zorder)
```

Comment:
I have a slight preference to inline (for less indirection):

```suggestion
        x_twins.sort(key=attrgetter("zorder"))
        y_twins.sort(key=attrgetter("zorder"))
```

Reply:
Further inlined.